File: programers_Lv2/k진수에서 소수 개수 구하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/92335
function solution(n, k) {
  let revBase = "";

  let num1 = n;
  while (num1 > 0) {
    let num2 = num1 % k;
    num1 = Math.floor(num1 / k);
    revBase += num2.toString();
  }
  revBase = revBase.split("").reverse().join("");
  const tempArr = revBase.split("0");

  const isPrime = (num) => {
    const start = 3;
    const end = parseInt(num ** 0.5) + 1;

    const range = Array.from(
      { length: end - start },
      (_, index) => index + start
    );

    for (let i of range) {
      if (num % i == 0) {
        return 0;
      }
    }
    return 1;
  };

  let count = 0;
  for (let i of tempArr) {
    let num = i.length != 0 ? Number(i) : 0;
    if (num < 2) {
      continue;
    }
    if (num == 2) {
      count++;
    } else {
      count += isPrime(num);
    }
  }
  return count;
}

console.log(solution(437674, 3));

File: programers_Lv2/k진수에서 소수 개수 구하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/92335
def solution(n, q):
    rev_base = ""

    while n > 0:
        n, mod = divmod(n, q)
        rev_base += str(mod)

    rev_base = rev_base[::-1]
    # 역순인 진수를 뒤집어 줘야 원래 변환 하고자하는 base가 출력

    temp_lst = rev_base.split("0")
    count = 0

    # 소수인지 판별하는 함수
    def is_prime(num):
        for i in range(
            3, int(num**0.5) + 1
        ):  # num만 작성한 경우 시간 초과 => int(num**0.5) + 1
            if num % i == 0:
                return 0
        return 1

    for i in temp_lst:
        num = int(i) if len(i) != 0 else 0
        if num < 2:
            continue
        if num == 2:
            count += 1
        else:
            count += is_prime(num)
    return count


print(solution(437674, 3))
